yargs.js

```diff
@@ -707,6 +713,10 @@ function Yargs (processArgs, cwd, parentRequire) {
         }
       }
 
+      // recommend a command if recommendCommands() has
+      // been enabled, and no commands were found to execute
+      if (recommendCommands && argv._.length) validation.recommendCommands(argv._[argv._.length - 1], handlerKeys)
```

Comment:
It looks like this is called even when `argv._` only contains valid commands or arguments (called as part of running the command); it just potentially doesn't have any commands to match against.